src/libs/Middleware/SaveResponseInOnyx.js

```diff
@@ -0,0 +1,32 @@
+import Onyx from 'react-native-onyx';
+import _ from 'underscore';
+
+/**
+ * @param {Promise} response
+ * @param {Object} request
+ * @returns {Promise}
+ */
+function SaveResponseInOnyx(response, request) {
+    return response
+        .then((responseData) => {
+            // We'll only save the onyxData, successData and failureData for the refactored commands
+            if (_.has(responseData, 'onyxData')) {
+                let data;
+                if (responseData.jsonCode === 200) {
+                    data = [
+                        ...request.successData,
+                        ...responseData.onyxData,
+                    ];
+                } else {
+                    data = [
+                        ...request.failureData,
+                        ...responseData.onyxData,
+                    ];
+                }
+                Onyx.update(data);
+            }
+            return responseData;
```

Comment:
Also, unsure if we want/need to return the response after `Onyx.update()` - no middleware comes after this so it's probably fine to have it or not and add it later.

Reply:
Yeah, I was thinking the same, but I think it's better to return the response at the end if we add more middleware functions after `SaveResponseInOnyx` in the future, otherwise, the next engineer that adds a new middleware will scratch their head a little while until they figure out why the response is empty (like me 😂)

Reply:
What?! But now you are an expert after this struggling 😂